### DISCUSSION TITLE
Why the method uses Activator.CreateInstance instead of directly "new Class()"?

### DISCUSSION THREAD

**Discussion by nkw** [jasontaylordev/CleanArchitecture#575] [Q&A]
In the method GetNotificationCorrespondingToDomainEvent in https://github.com/jasontaylordev/CleanArchitecture/blob/main/src/Infrastructure/Services/DomainEventService.cs, it has the following method,

```c#
private INotification GetNotificationCorrespondingToDomainEvent(DomainEvent domainEvent)
{
    return (INotification)Activator.CreateInstance(
        typeof(DomainEventNotification<>).MakeGenericType(domainEvent.GetType()), domainEvent)!;
}
```

How it compares with new DomainEventNotification(....)? BTW, is the "!" at the end needed?

```c#
private INotification GetNotificationCorrespondingToDomainEvent(DomainEvent domainEvent)
{
    return new DomainEventNotification<DomainEvent>(domainEvent)!;
}
```

**Comment by deceptiveSimplicity** [comment]
I think this is because you can't have different NotificationHandlers for different DomainEvents when all DomainEvents end up in the same `DomainEventNotification<DomainEvent>` class.
